lms/services/email_preferences.py

```diff
@@ -29,8 +35,37 @@ class EmailPrefs:
     sat: bool = True
     sun: bool = True
 
-    def days(self) -> dict:
-        return {key: value for key, value in asdict(self).items() if key in self.DAYS}
+    @staticmethod
+    def user_preferences_key_for_email_digest_date(datetime_: datetime) -> str:
+        return f"instructor_email_digests.days.{EmailPreferences.DAYS[datetime_.weekday()]}"
+
+    def unsubscribe_instructor_digest(self) -> Self:
+        self.mon = self.tue = self.wed = self.thu = self.fri = self.sat = self.sun = (
+            False
+        )
+        return self
+
+    def serialize(self) -> dict[str, bool]:
+        return {
+            "instructor_email_digests.days.mon": self.mon,
+            "instructor_email_digests.days.tue": self.tue,
+            "instructor_email_digests.days.wed": self.wed,
+            "instructor_email_digests.days.thu": self.thu,
+            "instructor_email_digests.days.fri": self.fri,
+            "instructor_email_digests.days.sat": self.sat,
+            "instructor_email_digests.days.sun": self.sun,
+        }
+
+    @classmethod
+    def from_user_preferences(cls, user_preferences: UserPreferences) -> Self:
+        return cls(
+            h_userid=user_preferences.h_userid,
+            **{
+                key[len(EMAIL_DIGEST_KEY_PREFIX) :]: value
```

Comment:
This only passes the values for which we do have non-default value and uses the default of the dataclass for the rest.
I keep this to keep that behaviour.